### ISSUE TITLE
Wazuh-app 3.10.2 and Kibana 7.4.2 no results available

### ISSUE THREAD

**Issue by hcv-target** [wazuh/wazuh-kibana-app#1894]
| Wazuh | Elastic |
| ----- | ------- |
| 3.10.2   | 7.x     |

**Description**
Following upgrade of elastic stack from 6.8.3 to 7.4.2, I see no data in the Wazuh Kibana app. I setup xpack users and roles per [this article](https://documentation.wazuh.com/3.8/user-manual/kibana-app/configure-xpack/defining-xpack-users.html) and my API is accessible. I followed the advice given [in this issue](https://github.com/wazuh/wazuh-kibana-app/issues/1867), adapting the paths for the 7.x release branch to obtain filebeat configuration and index template. I can confirm there is traffic on /var/ossec/logs/alerts/alerts.json, and this is the log path configured in the wazuh module for filebeat. It doesn't appear that filebeat is accessing the alerts.json, as the output of lsof /var/ossec/logs/alerts/alerts.json indicates only ossec-manager and ossec-analysisd are touching it. None of the fields in the wazuh-alerts template are visible in Discovery, but if you show hidden fields, they are all visible there.

I think the problem is, I don't have a wazuh module in /usr/share/filebeat/module.

Where do I get this? I've uninstalled wazuh-kibana-app and reinstalled it, but no joy.

Thanks again for this fantastic product and for the great support you guys provide.
-- 
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target,

Effectively, the Wazuh App doesn't start because you don't have the wazuh module for filebeat. Filebeat is the tool on the Wazuh server that securely forwards alerts and archived events to Elasticsearch. For this reason, it is impossible for alerts and events to appear.

Following these steps you can configure Filebeat:

- Download the Filebeat configuration file from the Wazuh repository. This is pre-configured to forward Wazuh alerts to Elasticsearch:

```
curl -so /etc/filebeat/filebeat.yml https://raw.githubusercontent.com/wazuh/wazuh/v3.10.2/extensions/filebeat/7.x/filebeat.yml
chmod go+r /etc/filebeat/filebeat.yml
```

- Download the Wazuh module for Filebeat:

```
curl -s https://packages.wazuh.com/3.x/filebeat/wazuh-filebeat-0.1.tar.gz | sudo tar -xvz -C /usr/share/filebeat/module
```

- Edit the file `/etc/filebeat/filebeat.yml` and replace YOUR_ELASTIC_SERVER_IP with the IP address or the hostname of the Elasticsearch server. For example:

```
output.elasticsearch.hosts: ['http://YOUR_ELASTIC_SERVER_IP:9200']
```

- Enable and start the Filebeat service:

_For Systemd_:

```
# systemctl daemon-reload
# systemctl enable filebeat.service
# systemctl start filebeat.service
```

_For SysV Init_:

```
# chkconfig --add filebeat
# service filebeat start
```

For more information:
https://documentation.wazuh.com/3.10/installation-guide/installing-wazuh-manager/linux/centos/wazuh_server_packages_centos.html#installing-filebeat

Let me know if you have any issues with the configuration

Regards.

**Comment by hcv-target** [comment]
Hi, @Joanes04 and thanks for your help. With the addition of the wazuh module for filebeat and some additional configuration, I now see daily indexes being created, and that seems to be passed through logstash, while I was expecting filebeat to be the one accessing the /var/ossec/logs/alerts/alerts.json. They are growing in size, so I'm going to assume they're being written to, but I still don't see events in wazuh-kibana-app. When viewing the Security Events > Discover tab, I still see no fields in the Available Fields list:
![Screen Shot 2019-11-14 at 11 07 28 AM](https://user-images.githubusercontent.com/11314969/68874320-00d24a80-06cf-11ea-94e5-c35325e57098.png)

...until I uncheck Hide missing fields:
![Screen Shot 2019-11-14 at 11 09 06 AM](https://user-images.githubusercontent.com/11314969/68874460-3bd47e00-06cf-11ea-9c91-7a0aeb76ded7.png)

Is it possible the dashboards did not get updated when I updated the plugin? I have uninstalled and reinstalled wazuh-kibana-app, and this did not have any effect.

Thanks again for any light you can shine on this.

Cheers,
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target, 

In order to keep helping you, I'd need some information from you: 

1. Can you pass me your **filebeat.yml** file? (_etc/filebeat/filebeat.yml_) 
2.  Why do you use Logstash? Do you need it for something specific? In most cases, filebeat would be enough
3. Can you make the request to ElasticSearch to consult the indexes and attach it? (**curl -X GET "ELASTIC_IP:9200/_cat/indices?v&pretty"**)
4. Can you make the request to ElasticSearch to consult the Wazuh template and attach it? (**curl -X GET "ELASTIC_IP:9200/_template/wazuh?pretty"**)

Please attach everything to the answer so I can review it.

Regards.

**Comment by hcv-target** [comment]
Hi, @Joanes04 please excuse the slow reply.

All requested info attached.
[_cat_indices.txt](https://github.com/wazuh/wazuh-kibana-app/files/3859609/_cat_indices.txt)
[_template.txt](https://github.com/wazuh/wazuh-kibana-app/files/3859610/_template.txt)
[filebeat.yml.txt](https://github.com/wazuh/wazuh-kibana-app/files/3859611/filebeat.yml.txt)

**Comment by Joanes04** [comment]
Hi @hcv-target,

I've checked that everything you've been through for me is correct. We're going to keep doing checks:

1. Can you confirm that the time range you have selected is wide?:

![Sin título](https://user-images.githubusercontent.com/9945959/69134844-70be4780-0ab8-11ea-83d4-0f0f701b8f70.png)

2. If it doesn't work with this range, try refreshing the index fields:

- Select `Management/Index Patterns/wazuh-alerts-3.x-*`

![Sin título](https://user-images.githubusercontent.com/9945959/69135275-28535980-0ab9-11ea-96f3-0f79583dd72c.png)

- Finally, click on `Refresh field list`

![Sin título](https://user-images.githubusercontent.com/9945959/69135453-65b7e700-0ab9-11ea-9791-503590405858.png)

I'm waiting for the results.
Regards.

**Comment by hcv-target** [comment]
Hi, @Joanes04 . Selecting a time range of any width shows no results. Refreshing the index pattern had no effect either. Thanks for continuing to look into this with me.

Cheers, 
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target,

We're gonna keep doing checks, but it's weird that you don't get results.

- Could you check the discover of Kibana to see if you have results outside of Wazuh? Your indexes are right, and the template is right too

- Let's check the alerts on one of your indexes. Please go to Dev Tools in Kibana, and make the following request: `GET wazuh-alerts-3.x-2019.11.18/_search?size=1`

![Sin título](https://user-images.githubusercontent.com/9945959/69164059-93b81e00-0aef-11ea-8499-8db6e7a463aa.png)

Please send me a screenshot. But remember, **HIDE** sensitive message information, such as IPs

Regards,

Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04 . Outside of Wazuh, Kibana is working as expected. Filebeat, Metricbeat and monitoring on all hosts is OK. Attached is screenshot as requested. I also  increased the size to 100 to see what else was going on besides osquery, and didn't see any other activity, FWIW.
<img width="1077" alt="Screen Shot 2019-11-20 at 7 17 58 AM" src="https://user-images.githubusercontent.com/11314969/69238684-9144db00-0b66-11ea-8255-dcbe68d28f41.png">

**Comment by Joanes04** [comment]
Hi @hcv-target,

It's very strange that it doesn't work for you, but I'm trying to reproduce its error and I can't get it. Forgive me!

- Can you send me the whole alert? In an attached document or something like that, because on the screen I can't see all the fields.

- Can you attach a screenshot of `Overview / Security events` with `date range = 1 year`?
Please check the filters applied, to make sure that it is not a problem of the filters the application applies implicitly.

![Sin título](https://user-images.githubusercontent.com/9945959/69247044-0f76a100-0baa-11ea-9e49-73bdce6f1a99.png)

![Sin título 2](https://user-images.githubusercontent.com/9945959/69247146-3cc34f00-0baa-11ea-86e1-ee82c3fbe1eb.png)

Regards!

**Comment by hcv-target** [comment]
Hi, @Joanes04. Full query result attached. 
[wazuh-alerts-search-result.txt](https://github.com/wazuh/wazuh-kibana-app/files/3870016/wazuh-alerts-search-result.txt)

Following the upgrade, when no data was appearing in my new indices based on the v3.10.2/7.4.2 index format, I deleted all previous wazuh indices in an attempt to 'start from scratch'. Thus, I have no data previous to the upgrade, so the second part of your request will have no meaning in this context. 

Thanks again for your help with this!

**Comment by Joanes04** [comment]
Hi @hcv-target,

In previous Elastic serch versions, the Elastic documents were indexed using the field @timestamp as the reference field. In Elastic 7.x, this field has become a reserved field and it is no longer manipulable. Wazuh now make use of field timestamp instead. I've been checking the alerts, and I see that **@timestramp** appears instead of **timestamp** as field.

1) Follow the guide of the following link for the migration:

https://documentation.wazuh.com/3.10/upgrade-guide/upgrading-elastic-stack/elastic_server_rolling_upgrade.html#field-migration-from-timestamp-to-timestamp

2) Can you attach a screenshot of index patterns of kibana panel:

![Sin título](https://user-images.githubusercontent.com/9945959/69348213-1bc83000-0c76-11ea-8ff1-9ae5a61ea4fd.png)

3) Check again the discover of Kibana and Wazuh, to see if you have any results.

4) Can you attach a screenshot of Overview / Security events with date range = 1 year? It's to check if you're using @timestamp or timestamp. I don't need you to have any data :)
 
Regards!

Alberto Andújar

**Comment by hcv-target** [comment]
Hi, @Joanes04 . If I am reading this right, this link discusses upgrading indices created using the previous index pattern under Elastic 6.8, doesn't it? All current indices were created using the new pattern under 7.4.2. I don't have any using the old pattern. Why would my indices created under 7.4.2 have '@timestamp' instead of 'timestamp'?

Nevertheless, I tried it, and it had no effect.
![image](https://user-images.githubusercontent.com/11314969/69379997-8a5dbb80-0c7f-11ea-8597-a788162d86a1.png)

Here are the index patterns
![Screen Shot 2019-11-21 at 4 45 58 PM](https://user-images.githubusercontent.com/11314969/69379429-76658a00-0c7e-11ea-8116-d8f9a04f5be0.png)

And here is the 1 year view. I presume you wanted to see discover. But I don't see any reference to either 'timestamp' or '@timestamp' in Dashboard either:
![image](https://user-images.githubusercontent.com/11314969/69380299-1cfe5a80-0c80-11ea-9352-4dab4deca326.png)

Thanks again!

**Comment by Joanes04** [comment]
Hi @hcv-target,

Sorry for the delay in response. We are trying to reproduce your error, but we have not succeeded. Let's continue with some checks to be able to find the error!

- Please attach the file found in the path: 
`/usr/share/filebeat/module/wazuh/alerts/ingest/pipeline.json`

- Can you check if you have running **logstash**? In previous versions, **logstash** had a filter that removes **timestramp**. You should use the command: 

```
# systemctl stop logstash
# systemctl daemon-reload
```
-  Can you attach a screenshot of the **wazuh-alerts*** index patterns of kibana panel:

![image](https://user-images.githubusercontent.com/9945959/69787342-a0f29e00-11bc-11ea-8051-69cd21079712.png)
![image](https://user-images.githubusercontent.com/9945959/69787368-aea82380-11bc-11ea-98d8-13828838e95a.png)

Regards!

Alberto Andújar

**Comment by hcv-target** [comment]
Hi, @Joanes04 no problem, I was on holiday last week so no time lost. Here is the pipeline.json:
[pipeline.json.txt](https://github.com/wazuh/wazuh-kibana-app/files/3912295/pipeline.json.txt)

Logstash has been running, but I did the daemon-reload as requested. There doesn't seem to be any difference in behavior.

Here is the index pattern detail screenshot:
![Screen Shot 2019-12-02 at 11 59 27 AM](https://user-images.githubusercontent.com/11314969/69982116-b5f36800-1501-11ea-9033-b468a05b57ce.png)

Once again, thanks for working through this with me.

Best regards,
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target ,

Please let us check that the filebeat installation and configuration is correct. To do this, run (on the machine where you have installed filebeat) the following command: 

`filebeat test output`

Please, can you attach the output returned by the command to a file?

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04 , here is the output from filebeat test output:

[filebeat_test_output.txt](https://github.com/wazuh/wazuh-kibana-app/files/3922494/filebeat_test_output.txt)

Thanks!

**Comment by Joanes04** [comment]
Hi @hcv-target,

Your filebeat settings looks good. One possible solution is to regenerate all the kibana indices. 
**IMPORTANT, these steps remove all the information you have in kibana. Both the indexes, the spaces...** If you have data that you would like to keep, please let me know and I will explain how to do the next step by step through the terminal, so that you do not lose the kibana information.

If you can remove all your information without problems, you can follow the steps below:

- Go to **Dev Tools** of kibana panel:

![image](https://user-images.githubusercontent.com/9945959/70157271-969d3c00-16b5-11ea-8156-9cd9bd63f921.png)

- Now executes the following query `DELETE .kibana*`

![image](https://user-images.githubusercontent.com/9945959/70157432-dfed8b80-16b5-11ea-93e3-98e89f365aea.png)

- And you need restart Kibana

`sudo systemctl restart kibana.service`

With these steps we will regenerate all the indexes of kibana, Index patterns, Spaces... If you need to save the information you have so far, please let me know. 

Best regards,
Alberto

**Comment by hcv-target** [comment]
Thanks, @Joanes04 . Funny, when I sent this morning's reply, I was considering deleting .kibana*, but decided to wait for your response.

I'm not worried about losing data - if the indices are the problem, keeping them isn't going to solve the problem, so... away they go.

I've deleted .kibana*, done metricbeat setup and filebeat setup, and these seem to be working as expected. 

Wazuh-kibana app is still telling me 
`There are no results for selected time range. Try another one.`

Now, when I try to select a different time range, the time selector disappears altogether and I get a javascript error:
`TypeError: Cannot read property 'includes' of null
    at getDateMode (vendors.bundle.dll.js:111)
    at isRelativeToNow (vendors.bundle.dll.js:48)
    at prettyDuration (vendors.bundle.dll.js:48)
    at vendors.bundle.dll.js:208
    at Array.map (<anonymous>)
    at EuiRecentlyUsed (vendors.bundle.dll.js:208)
    at hg (vendors.bundle.dll.js:200)
    at Tg (vendors.bundle.dll.js:200)
    at bi (vendors.bundle.dll.js:200)
    at ci (vendors.bundle.dll.js:200)
qh @ vendors.bundle.dll.js:200
Ch.c.callback @ vendors.bundle.dll.js:200
ih @ vendors.bundle.dll.js:200
hh @ vendors.bundle.dll.js:200
Vh @ vendors.bundle.dll.js:200
Zh @ vendors.bundle.dll.js:200
(anonymous) @ vendors.bundle.dll.js:200
exports.unstable_runWithPriority @ vendors.bundle.dll.js:151
Fi @ vendors.bundle.dll.js:200
Di @ vendors.bundle.dll.js:200
Yh @ vendors.bundle.dll.js:200
Ii @ vendors.bundle.dll.js:200
Cd @ vendors.bundle.dll.js:200
vendors.bundle.dll.js:111 Uncaught TypeError: Cannot read property 'includes' of null
    at getDateMode (vendors.bundle.dll.js:111)
    at isRelativeToNow (vendors.bundle.dll.js:48)
    at prettyDuration (vendors.bundle.dll.js:48)
    at vendors.bundle.dll.js:208
    at Array.map (<anonymous>)
    at EuiRecentlyUsed (vendors.bundle.dll.js:208)
    at hg (vendors.bundle.dll.js:200)
    at Tg (vendors.bundle.dll.js:200)
    at bi (vendors.bundle.dll.js:200)
    at ci (vendors.bundle.dll.js:200)`

**Comment by Joanes04** [comment]
Hi @hcv-target,

We are closer to solving the problem :). You've restarted kibana after eliminating everything, right?
If you have restarted kibana, can you check the index pattern detail, if the filter field is **timestamp**?

Can you pass me the index-pattern you have on wazuh? To do this run from the Dev Tools of kibana the following query and pass me the reply please:

```
// QUERY

GET .kibana*/_search
{
  "query": {
    "wildcard": {
      "index-pattern.title": {
        "value": "wazuh*"
      }
    }
  },
  "_source": ["index-pattern.title","index-pattern.timeFieldName"]
}`

```

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04, thanks for the confidence boost ;)

After revising the query to use POST instead of GET (honest mistake), here is my output:

[wazuh-index-patterns.txt](https://github.com/wazuh/wazuh-kibana-app/files/3928061/wazuh-index-patterns.txt)

Perhaps of note is the difference between the filebeat-* and metricbeat-* index patterns:
![Screen Shot 2019-12-05 at 12 08 14 PM](https://user-images.githubusercontent.com/11314969/70257355-1e3b8700-1758-11ea-9ca9-7a9f98394c82.png)
![Screen Shot 2019-12-05 at 12 08 31 PM](https://user-images.githubusercontent.com/11314969/70257370-24316800-1758-11ea-9385-b156f138c932.png)

and that for wazuh-alerts-3.x-*:
![Screen Shot 2019-12-05 at 12 08 52 PM](https://user-images.githubusercontent.com/11314969/70257408-357a7480-1758-11ea-9fc9-f532693cbe37.png)

...or is this the expected behavior?

Thanks again, sir!
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target !

The first thing is to tell you that it is the expected behavior:

> **@timestamp**
> Date/time when the event originated. This is the date/time extracted from the event, typically representing when the event was generated by the source. If the event source has no original timestamp, this value is typically populated by the first time the event was received by the pipeline. Required field for all events.
> type: date
> example: 2016-05-23T08:05:34.853Z
> required: True

> **timestamp**
> Date/time when the alert is created. This is the date/time when the event took place.
> type: date
> example: 2016-05-23T08:05:34.853Z
> required: True

Your wazuh-index-patterns looks right. So the problem doesn't come from this.

We're going to keep testing possible causes. Please try to use Wazuh with the browser incognito, as it may be that something is stored in the cache that is not working at all well.

Thank you!

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04 and thanks for the info. I've tried the wazuh app in Chrome Incognito, Firefox and Safari with identical results; I always use Chrome and almost never Firefox or Safari, so I'm confident it's not cache-related. Hope to hear more from you soon!

Cheers,
Ken

**Comment by hcv-target** [comment]
Hi, @Joanes04 , I took a step back today and went back to basics on this issue. One thing that came to mind was I've never seen lsof /var/ossec/logs/alerts/alerts.json return any instance where user 'filebeat' was indicated as reading it. 

This led me to discover I have no user 'filebeat' on this system. My question then is "shouldn't I have user 'filebeat', and if so, when would it have been created?". I have users 'logstash', 'kibana', 'elasticsearch', but no filebeat. Is this as expected?

**Comment by Joanes04** [comment]
Hi @hcv-target ,

The truth is that you don't need a Filebeat user, so your configuration is correct again.
The lsof must get a result like this:

![image](https://user-images.githubusercontent.com/9945959/70974119-53749d00-20a7-11ea-96f2-4f915919f458.png)

Forgive me for waiting so long, but as hard as I try, I can't reproduce your mistake. Let's check the current state again.

- Can you pass me the index-pattern you have on wazuh? To do this run from the Dev Tools of kibana the following query and pass me the reply please:

`GET .kibana*/_search`

- Can you attach a full alert of the **most recent index**, the **log error of elasticsearch** and the **log of Wazuh App** ?

- After this, let's reinstall the wazuh plugin as follows:


Stop Kibana

- Systemd:

```
systemctl stop kibana
```

- SysV Init:

```
service kibana stop
```

Remove the app using kibana-plugin tool

```
/usr/share/kibana/bin/kibana-plugin remove wazuh
```

Remove generated bundles

```
rm -rf /usr/share/kibana/optimize/bundles
```

Update file permissions. This will avoid several errors prior to updating the app:

```
chown -R kibana:kibana /usr/share/kibana/optimize
chown -R kibana:kibana /usr/share/kibana/plugins
```

Install the app

- With sudo:

```
sudo -u kibana /usr/share/kibana/bin/kibana-plugin install https://packages.wazuh.com/wazuhapp/wazuhapp-3.10.2_7.5.0.zip
```

- Without sudo:

```
su -c '/usr/share/kibana/bin/kibana-plugin install https://packages.wazuh.com/wazuhapp/wazuhapp-3.10.2_7.5.0.zip' kibana
```

Restart Kibana

- Systemd:

```
systemctl restart kibana
```
 
- SysV Init: 
 
``` 
service kibana restart 
``` 
 
Best regards,
Alberto

**Comment by hcv-target** [comment]
Good morning, @Joanes04 . When I lsof /var/ossec/logs/alerts/alerts.json, I get
![image](https://user-images.githubusercontent.com/11314969/70994142-8a44b600-209b-11ea-9405-2d3f65d7a767.png)

No mention of filebeat. How can I check if filebeat is running the wazuh module successfully?

I'll post back when I've reinstalled the app, but I don't think this is going to make a difference, as I've already done so more than once since I started this thread.

Thanks again!

**Comment by hcv-target** [comment]
@Joanes04 , here's the output from GET .kibana*/_search:

[kibana_search.json.txt](https://github.com/wazuh/wazuh-kibana-app/files/3975477/kibana_search.json.txt)

I reinstalled the app as requested, with no change to this issue.

Thanks for everything...

**Comment by Joanes04** [comment]
Hi @hcv-target ,

I see you're still using logstash, let's try one thing:

- First stop logstash:

`sudo systemctl stop logstash.service
`
- Check the status of Filebeat

`systemctl status filebeat.service
`
- If the service is stopped, try to start it with :

`sudo systemctl start filebeat.service
`
- I'll give you an example of a `filebeat.yml` file that works correctly. Please check that your `filebeat.yml` is just as correct:

```
# Wazuh - Filebeat configuration file
filebeat.modules:
  - module: wazuh
    alerts:
      enabled: true
    archives:
      enabled: false

setup.template.json.enabled: true
setup.template.json.path: '/etc/filebeat/wazuh-template.json'
setup.template.json.name: 'wazuh'
setup.template.overwrite: true
setup.ilm.enabled: false

output.elasticsearch.hosts: ['http://YOUR_ELASTIC_SERVER_IP:9200']

```

- The filebeat.yml file is in the following path:

`/etc/filebeat/filebeat.yml`

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04 . After stopping logstash and revising /etc/filebeat/modules.d/wazuh.yml to match the above.

lsof /var/ossec/logs/alerts/alerts.json now returns
![image](https://user-images.githubusercontent.com/11314969/71115303-a03b9f00-219f-11ea-8f33-3c7dd0811cbe.png)

I now receive the following error in the wazuh-app startup screen:

Health Check. 3002 - [cluster_block_exception] index [.wazuh] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)]; (/api/check-stored-api)

When I revised the wazuh.yml, I removed the login credentials for filebeat that I had placed there, as your example had none. I also specified the correct protocol in output.elasticsearch.hosts: ['**https**://ELASTIC_SERVER_IP:9200']

I look forward to your response.
Best regards,
Ken

**Comment by hcv-target** [comment]
@Joanes04 , I figured out the Health Check 3002 - my disk had filled up while debugging was on in filebeat. Once I made some room I was able to check wazuh-app again, and no joy.

**Comment by Joanes04** [comment]
Hi @hcv-target ,

This error may be caused due to exceeding _flood_stage disk watermark_ on the ES host.
The flood stage watermark is 95% by default, so on a 1TB drive you need at least 50GB of free space or Elasticsearch will put itself into read-only mode. as mentioned in Elastic docs :(https://www.elastic.co/guide/en/elasticsearch/reference/current/disk-allocator.html) 
 

> Elasticsearch enforces a read-only index block (_index.blocks.read_only_allow_delete_) on every index that has one or more shards allocated on the node that has at least one disk exceeding the flood stageThis is a last resort to prevent nodes from running out of disk space. The index block must be released manually once there is enough disk space available to allow indexing operations to continue.

You can unlock these blocked indices with this command:
```
PUT _all/_settings
{
  "index.blocks.read_only_allow_delete": null
}
```

These watermark settings can be updated, for example we can set disk watermark as byte value instead of percentage value:
```
PUT _cluster/settings
{
  "transient": {
    "cluster.routing.allocation.disk.watermark.low": "100gb",
    "cluster.routing.allocation.disk.watermark.high": "50gb",
    "cluster.routing.allocation.disk.watermark.flood_stage": "10gb",
  }
}
```

1.- Can you pass me your filebeat.yml? It's to check that it's correct. (`/etc/filebeat/filebeat.yml`)

2.- Can you do a filebeat test and attach the result? You have to execute the following command:
`sudo filebeat test output`

3.- Can you check the filebeat status and attach the result? You have to execute the following command:
` systemctl status filebeat`

4.- Let's check the alerts on one of your indexes. Please go to Dev Tools in Kibana, and make the following request: `GET wazuh-alerts-3.x-2019.12.18/_search`

![image](https://user-images.githubusercontent.com/9945959/71155405-e6dfd680-223d-11ea-9cc8-756ca9579187.png)

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hello, @Joanes04 I hope you had a fine Holiday season. Happy New Year. By way of response, here are the requested files:
[get wazuh-alerts-3.x-2020.01.02_search.txt](https://github.com/wazuh/wazuh-kibana-app/files/4017519/get.wazuh-alerts-3.x-2020.01.02_search.txt)
[systemctl status filebeat.txt](https://github.com/wazuh/wazuh-kibana-app/files/4017520/systemctl.status.filebeat.txt)
[filebeat test output.txt](https://github.com/wazuh/wazuh-kibana-app/files/4017521/filebeat.test.output.txt)
[filebeat.yml.txt](https://github.com/wazuh/wazuh-kibana-app/files/4017523/filebeat.yml.txt)

With logstash stopped, indices do not get created. Filebeat is still not correctly reading alerts.json. For these tests, I had to enable logstash so the _search would return results.

The previous 3002 error was cleared when I freed some disk space.

Now when loading the Wazuh app I get:

Health Check. 3002 - Error reading file (/api/check-stored-api)

I get a message about setting up the API credentials, but they work fine when tested from within the app.

Hope 2020 is a better year for fixing this than 2019 was. Thanks for sticking with it. Cheers,
-- 
Ken

**Comment by Joanes04** [comment]
Hi @hcv-target!
Happy New Year! Sorry for the delay in response,  I was on holiday last week.
Let's try a basic Filebeat configuration. To do so, please follow the steps below:

1) First save a copy of your **Filebeat.yml** file
2) I attach a Filebeat.yml file created by default, please use this (Remember to change the IP)
[filebeat.yml.zip](https://github.com/wazuh/wazuh-kibana-app/files/4029516/filebeat.yml.zip)
3) Stops the logstash service
```
# systemctl stop logstash
# systemctl daemon-reload
```
4) Now let's delete all the Wazuh indexes. **WARNING**: Please note that you will delete all the indexes. Please let me know if removing all your indices is not possible for you.
a)  Go to **Dev Tools** of Kibana panel:

![image](https://user-images.githubusercontent.com/9945959/71880864-7f78c080-3131-11ea-8fd9-ae4917e06fc4.png)

b) Now executes the following query:
`DELETE wazuh-alerts*`

5) Finally, let's remove the Wazuh temlates.
a) Go to **Dev Tools** of Kibana panel again
b) Now execute the following query:
`GET _cat/templates`
c) Find the wazuh-alerts template* and run the following query:
`DELETE _template/NAME_OF_TEMPLATE`

6) Now you can restart Filebeat and test again if it works
```
# systemctl restart logstash
# systemctl daemon-reload
```

Best regards,
Alberto

**Comment by hcv-target** [comment]
Hi, @Joanes04 no worries, sir. I've been swamped with my actual work, so haven't gotten back to you until now.

My filebeat.yml is already identical to your example, so I don't need to change anything there.

I've followed you up to point 6, where you talk about first restarting filebeat, then logstash. Assuming you meant filebeat, after restarting filebeat no index is created as Filebeat is still not generating indices. I have verified the template was created and has both timestamp and @timestamp fields as previously seen.

Thanks again, look forward to hearing next from you.

Ken

**Comment by pablotr9** [comment]
Hi @hcv-target ,

Let me try to help you! We will do some checks, sorry if you already did all these steps (I know this will be a little bit repetitive) but this took so long and I need to see the current state of your environment.

1. Please tell me your current version of Wazuh and Elasticsearch/Kibana/Filebeat.
2. Make sure that Logstash is stopped, we will only use Filebeat. (systemctl stop logstash - or uninstall it if it's not being used)
3. Please share with me your Filebeat configuration file (`cat /etc/filebeat/filebeat.yml`)
4. Now test the output of Filebeat to your Elasticsearch (`filebeat test output`)
5. Now let's move to Elasticsearch and check its status, you can all this commands in the Kibana Dev Tools:
    - Check Elasticsearch indices: `GET _cat/indices/`
    - Check Templates `GET _cat/templates`
    - Check Wazuh alerts template: `GET _template/wazuh`
    - Check ingest pipelines (only wa: `GET _ingest/pipeline` 
    - Check the structure of one alert (change the index name to an existing one):
 `GET wazuh-alerts-3.x-2020.01.23/_search?size=1`
6. Let's move to check the status of Wazuh:
    - We will enable the debug mode: Edit config.js file. (/var/ossec/api/configuration/config.js) and replace this line:
`config.logs = "info";`
with
`config.logs = "debug";`
Now restart Wazuh manager and Wazuh API.
    - Now that you have the Debug mode enabled, let's check the status with this request:
`curl -u user:password  http://WAZUH-IP:55000/manager/info?pretty`
    - If there is an error in that request, please check Wazuh API logs in this file:
     `tail -n 50 /var/ossec/logs/api.log`


I think that is enough to know the status of your environment, please paste the output of all the different steps in a file and attach those files in your reply.

Kind Regards,
Pablo Torres

**Comment by hcv-target** [comment]
Hi, @pablotr9 . I have been unable to return to this until now. 

Since my last contact on this ticket, accessing the wazuh-app now returns the error:

No template found for the selected index-pattern.
Health Check. 3002 - Error reading file (/api/check-stored-api)

Here are my responses to your requests:

[filebeat.yml.txt](https://github.com/wazuh/wazuh-kibana-app/files/4155946/filebeat.yml.txt)
[filebeat_test_output.txt](https://github.com/wazuh/wazuh-kibana-app/files/4155954/filebeat_test_output.txt)
[_cat_indices.txt](https://github.com/wazuh/wazuh-kibana-app/files/4155957/_cat_indices.txt)
[_cat_templates.txt](https://github.com/wazuh/wazuh-kibana-app/files/4155962/_cat_templates.txt)

GET _template/wazuh returns an empty set because the indices are not being created by filebeat.

[_ingest_pipeline.txt](https://github.com/wazuh/wazuh-kibana-app/files/4155974/_ingest_pipeline.txt)

Once again, there are no alerts being created because filebeat won't write the index, so I can't supply the alert structure.

With config.logs = "debug", I discovered the user I was using to write to the index was not able to authenticate. Once I used the correct credentials for the API, the output from the curl was:

{
   "error": 1005,
   "message": "Error reading file"
}

Looking forward to hearing what you recommend. 

Best regards,
Ken

**Comment by pablotr9** [comment]
Hi @hcv-target ,

Ok thanks for the output, let's go step by step.

------------

Let's start with the `Error reading file` you are getting in your Wazuh API. Now that the debug mode is enabled we need to check the Wazuh API log file (`/var/ossec/logs/api.log`). 
Please do this request
`curl -u user:password http://WAZUH-IP:55000/manager/info?pretty`
The output of this command won't give us any extra information but the log file will. Now please check the Wazuh API log file
`tail -n 150 /var/ossec/logs/api.log` 
The output should show what file is giving you problems, probably a permissions problem. Attach the output of that last command so I can give you further assistance.


------------

Let's move now to your Elasticsearch problem.
Our Filebeat module contains all the information needed to read from our `alerts.json` file, process the data, apply our template and ship those logs to the specific `wazuh-alerts-3.x-<date>` index.
I can see in your Filebeat configuration that there is no Wazuh-related configuration so it's impossible to have wazuh-alerts indices. What we will do now is correctly configure Filebeat:
1. Stop Filebeat:
`systemctl stop filebeat`
2. We will rename your current `filebeat.yml` so we make sure that you do not lose any of your configuration.
`mv /etc/filebeat/filebeat.yml /etc/filebeat/filebeat.yml.old`
You can now find your old configuration in `/etc/filebeat/filebeat.yml.old` in case you want to restore it.
3. Let's now create a basic Filebeat configuration with our module enabled:
Create a file in `filebeat.yml` in /etc/filebeat/ and add this to it:
```
filebeat.modules:
  - module: wazuh
    alerts:
      enabled: true
    archives:
      enabled: true

setup.template.json.enabled: true
setup.template.json.path: '/etc/filebeat/wazuh-template.json'
setup.template.json.name: 'wazuh'
setup.template.overwrite: true
setup.ilm.enabled: false


#This is the same output configuration you had in your old filebeat.yml
output.elasticsearch:
  # Array of hosts to connect to.
  hosts: ["[ELASTIC_IP)]:9200"]
  ssl.certificate: "/etc/filebeat/certs/wazuh-manager-remote.crt"
  ssl.key: "/etc/filebeat/certs/wazuh-manager-remote.key"
  ssl.certificate_authorities: ["/etc/filebeat/certs/ca/ca-remote.crt"]

  # Enabled ilm (beta) to use index lifecycle management instead daily indices.
  ilm.enabled: true

  # Optional protocol and basic auth credentials.
  protocol: "https"
  username: "elastic"
  password: "[ELASTIC_PASS]"
```
4. This will work only if the module is installed and the wazuh-template has been downloaded, so please check that everything is installed:
   - Check that the Wazuh module directory exists: `ls /usr/share/filebeat/module/wazuh` 
   - Check that wazuh-template is downloaded: `cat /etc/filebeat/wazuh-template.json`
 If any of them is not installed, please check this guide to install them https://documentation.wazuh.com/3.11/installation-guide/installing-wazuh-manager/linux/centos/wazuh_server_packages_centos.html#installing-filebeat  (if you need any help, please let me know)

5. Now restart Filebeat and run this command:
`filebeat setup --pipelines`

Please check now if `wazuh-alerts-3*` are being populated (make sure that you generate alerts):
`GET _cat/indices/wazuh-alert*` 

I hope it helps, please let me know if I have to clarify something or need any extra help in any of the mentioned steps!

Kind Regards,
Pablo Torres

**Comment by hcv-target** [comment]
Hi, @pablotr9 . The wazuh-app configuration is located in /etc/filebeat/modules.d/wazuh.yml and is loaded as a module. Is this not the correct method?

Thanks.

**Comment by pablotr9** [comment]
Hi @hcv-target ,

Yes, that's a different approach, but we do not provide any `wazuh.yml` for that purpose, in our documentation is explained how to import the module directly in the `filebeat.yml` as you can see in my previous reply step 3, that is a basic Filebeat configuration example to import only the Wazuh module. Regarding the wazuh.yml, did you create it by yourself? could you please share the content of that file (/etc/filebeat/modules.d/wazuh.yml ) with me so I can check it out?

Regarding the `Error reading file`, are you still having trouble with it or is it already fixed?
And what about the steps of my last reply, did you check that everything is correctly installed (Wazuh module) and configurated? Our Wazuh module is responsible for sending the Wazuh template and Ingest pipeline to Elasticsearch.

Kind Regards,
Pablo Torres

**Comment by hcv-target** [comment]
Hi, @pablotr9 . Thanks for the response.

We set up the app using the wazuh-filebeat module at the direction of @Joanes04 back in November - please see [above](https://github.com/wazuh/wazuh-kibana-app/issues/1894#issuecomment-553331120). This configuration was provided by your team at that time, and is installed as directed.

Here is my /etc/filebeat/modules.d/wazuh.yml:

> [wazuh.yml.txt](https://github.com/wazuh/wazuh-kibana-app/files/4215151/wazuh.yml.txt)

No wazuh-alerts* indices are created after restarting filebeat, wazuh-manager, wazuh-api.

Upon visiting my-kibana-host.org:5601/app/wazuh, the health check fails with:

No template found for the selected index-pattern.
Health Check. 3002 - Error reading file (/api/check-stored-api)

A trace from /var/ossec/logs/api.log, with debug level output:

> WazuhAPI 2020-02-17 12:25:58 wazuh: <ELASTIC_IP> GET /manager/info
> WazuhAPI 2020-02-17 12:25:58 wazuh: CMD - Command: /var/ossec/framework/python/bin/python3 args:/var/ossec/api/models/wazuh-api.py stdin:{"function":"/manager/info","arguments":{"wait_for_complete":false}}
> WazuhAPI 2020-02-17 12:25:59 wazuh: CMD - Exit code: 0
> WazuhAPI 2020-02-17 12:25:59 wazuh: CMD - STDOUT:
> {"message": "Error reading file: /var/ossec/ruleset/VERSION", "error": 1005}
> 
> WazuhAPI 2020-02-17 12:25:59 wazuh: CMD - STDOUT: 77 bytes
> WazuhAPI 2020-02-17 12:25:59 wazuh: Error reading file: /var/ossec/ruleset/VERSION
> WazuhAPI 2020-02-17 12:25:59 wazuh: <ELASTIC_IP> GET /manager/info?pretty - 200 - error: '1005'.

If you click 'Go to App', you get:

> Wazuh App: Please set up Wazuh API credentials.

But the credentials are set up, and testing the connection is successful:

> Settings. Connection success

Upon examining /var/ossec/ruleset/VERSION, it had root:root permissions. I did chown root:ossec /var/ossec/ruleset/VERSION and now the /manager/info looks right, and loading the wazuh-kibana app results in:

> No template found for the selected index-pattern.

Clicking 'Go to App' now allows management of the app. Obviously we're still not creating indices, because we don't have a template for the index pattern.

I then did filebeat setup --modules wazuh and restarted filebeat, wazuh-manager and wazuh-api, just in case that would knock it loose, but it didn't.

I then moved the contents of /etc/filebeat/modules.d/wazuh.yml into /etc/filebeat/filebeat.yml, restarted filebeat, and now we have indices being generated and I am finally able to see my alerts in the kibana app. 

In retrospect, I think It was unclear to me that the wazuh filebeat module wasn't designed to work like other filebeat modules, with its config in /modules.d/. I think I can be excused for assuming it should be configured there; after all, that's where filebeat modules go, and you've labeled this a module.

Anyway, it's finally working, so I'm closing this issue at long last.

Thanks @pablotr9 and @Joanes04 for all your help.

**Comment by sadewale3** [comment]
Had same issue but after reading the last comment with resolution i was able to correct my mistake. 

I too made a wrong assumption about how the wazuh filebeat module was meant to be setup. 

good work troubleshooting guys.
